commit b6b20c84e8229b226906413b663faeb28b81297d
Author: Jonathan Liu <net147@gmail.com>
Date:   Thu Sep 11 00:47:46 2014 +0200

    Fix crash on quit if KDE wallet subsystem is disabled
    
    REVIEW: 120097

diff --git a/krfb/invitationsrfbserver.cpp b/krfb/invitationsrfbserver.cpp
index f144fb6..386fed5 100644
--- a/krfb/invitationsrfbserver.cpp
+++ b/krfb/invitationsrfbserver.cpp
@@ -54,8 +54,10 @@ void InvitationsRfbServer::init()
 
     instance->m_wallet = Wallet::openWallet(
             Wallet::NetworkWallet(), 0, Wallet::Asynchronous);
-    connect(instance->m_wallet, SIGNAL(walletOpened(bool)),
-            instance, SLOT(walletOpened(bool)));
+    if(instance->m_wallet) {
+        connect(instance->m_wallet, SIGNAL(walletOpened(bool)),
+                instance, SLOT(walletOpened(bool)));
+    }
 }
 
 const QString& InvitationsRfbServer::desktopPassword() const
@@ -121,7 +123,7 @@ InvitationsRfbServer::~InvitationsRfbServer()
     KSharedConfigPtr config = KGlobal::config();
     KConfigGroup krfbConfig(config,"Security");
     krfbConfig.writeEntry("allowUnattendedAccess",m_allowUnattendedAccess);
-    if(m_wallet->isOpen()) {
+    if(m_wallet && m_wallet->isOpen()) {
 
          if( (m_wallet->currentFolder()=="krfb") ||
                  ((m_wallet->hasFolder("krfb") || m_wallet->createFolder("krfb")) &&
@@ -150,6 +152,7 @@ void InvitationsRfbServer::walletOpened(bool opened)
 {
     QString desktopPassword;
     QString unattendedPassword;
+    Q_ASSERT(m_wallet);
     if( opened &&
             ( m_wallet->hasFolder("krfb") || m_wallet->createFolder("krfb") ) &&
             m_wallet->setFolder("krfb") ) {
